src/olympia/devhub/file_validation_annotations.py

```diff
@@ -65,226 +61,35 @@ def annotate_legacy_addon_restrictions(path, results, parsed_data, error=True):
-                                '\'name/value\'.'),
-                            description=[
-                                'Param elements in the OpenSearch '
-                                'provider must include a name and a '
-                                'value attribute.'])
+def annotate_search_plugin_restriction(results, file_path, channel):
+    """
+    Annotate validation results to restrict uploads of OpenSearch plugins
 
-                    param_value = (
-                        param.attributes['value'].value if
-                        'value' in param.attributes.keys() else '')
+    https://github.com/mozilla/addons-server/issues/12462
 
-                    if param_value.count('{searchTerms}'):
-                        found_template = True
+    Once this has settled for a while we may want to merge this with
+    `annotate_legacy_addon_restrictions`
+    """
+    if not file_path.endswith('.xml'):
+        return
 
-            # If the template still hasn't been found...
-            if not found_template:
-                tpl = url.attributes['template'].value
-                insert_validation_message(
-                    results,
-                    message=(
-                        'OpenSearch: <Url> element missing template '
-                        'placeholder.'),
-                    description=[
-                        '`<Url>` elements of OpenSearch providers '
-                        'must include a template attribute or specify a '
-                        'placeholder with `{searchTerms}`.',
-                        'Missing template: %s' % tpl])
+    # We can be broad here. Search plugins are not validated through this
+    # path and as of right now (Jan 2019) there aren't any legacy type
+    # add-ons allowed to submit anymore.
+    msg = ugettext(
+        u'Open Search add-ons are {blog_link_open}no longer supported on AMO'
+        u'{blog_link_close}. You can create a {doc_link_open}search extension '
+        u'instead{doc_link_close}.').format(
+            blog_link_open=(
+                '<a href="https://blog.mozilla.org/addons/2019/10/15/'
+                'search-engine-add-ons-to-be-removed-from-addons-mozilla-org/'
+                '">'),
+            blog_link_close='</a>',
+            doc_link_open=(
+                '<a href="https://developer.mozilla.org/docs/Mozilla/'
+                'Add-ons/WebExtensions/Your_first_WebExtension">'),
+            doc_link_close='</a>')
 
-    # Make sure there are no updateURL elements
-    if dom.getElementsByTagName('updateURL'):
-        insert_validation_message(
-            results,
-            message=(
-                'OpenSearch: <updateURL> elements are banned in OpenSearch '
-                'providers.'),
-            description=[
-                'OpenSearch providers may not contain <updateURL> elements.'])
+    insert_validation_message(
+        results, type_='error', message=msg, msg_id='legacy_addons_unsupported'
```

Comment:
we already use `legacy_addons_unsupported` as the `msg_id` in `annotate_legacy_addon_restrictions`, so do we want a different id?  (I'm not sure exactly what we use the msg_ids for

Reply:
Fixed in 44dd880928